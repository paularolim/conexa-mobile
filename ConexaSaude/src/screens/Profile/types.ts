export interface ProfileViewProps {
  name: string;
  email: string;
  handleLogout: () => void;
}
