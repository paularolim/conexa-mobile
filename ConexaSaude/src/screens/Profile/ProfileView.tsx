import { useTheme } from 'styled-components/native';

import { Divider } from '@components/Divider';
import { Icon } from '@components/Icon';

import {
  AuxiliaryLabel,
  Avatar,
  Center,
  Container,
  Header,
  MenuContainer,
  MenuItem,
  NameLabel,
} from './styles';
import { ProfileViewProps } from './types';

export function ProfileView({ name, email, handleLogout }: ProfileViewProps) {
  const { colors } = useTheme();

  return (
    <Container>
      <Header>
        <Avatar>
          <Icon name="user" width={80} height={80} fill={colors.pink} />
        </Avatar>
        <Center>
          <NameLabel>{name}</NameLabel>
          <AuxiliaryLabel>{email}</AuxiliaryLabel>
        </Center>
      </Header>

      <Divider />

      <MenuContainer>
        <MenuItem onPress={handleLogout}>
          <Icon name="sign-out" width={32} height={32} fill={colors.black} />
          <AuxiliaryLabel>Sair</AuxiliaryLabel>
        </MenuItem>
      </MenuContainer>
    </Container>
  );
}
