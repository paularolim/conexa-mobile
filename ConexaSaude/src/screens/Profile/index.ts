export * from './Profile';
export * from './ProfileView';
export * from './types';
export * from './useProfileViewModel';
