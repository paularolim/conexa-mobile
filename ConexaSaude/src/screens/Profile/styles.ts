import { TouchableOpacity, View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(View)`
  padding: 24px;
  gap: 24px;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  flex: 1;
`;

export const Header = styled(View)`
  gap: 16px;
  align-items: center;
`;

export const Center = styled(View)`
  align-items: center;
`;

export const Avatar = styled(View)`
  background-color: ${({ theme }) => theme.colors.grayLight};
  width: 80px;
  height: 80px;
  border-radius: 40px;
  overflow: hidden;
`;

export const NameLabel = styled(Text)`
  font-size: 22px;
  font-weight: bold;
`;

export const AuxiliaryLabel = styled(Text)`
  font-size: 18px;
`;

export const MenuContainer = styled(View)`
  width: 100%;
  gap: 8px;
`;

export const MenuItem = styled(TouchableOpacity)`
  flex-direction: row;
  align-items: center;
  gap: 8px;
`;
