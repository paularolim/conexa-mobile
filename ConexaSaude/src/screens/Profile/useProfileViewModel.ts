import { useAuth } from '@hooks/useAuth';

export function useProfileViewModel() {
  const { name, email, handleLogout } = useAuth();

  return { name: name || '', email: email || '', handleLogout };
}
