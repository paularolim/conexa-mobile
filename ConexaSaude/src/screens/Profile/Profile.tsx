import { ProfileView } from './ProfileView';
import { useProfileViewModel } from './useProfileViewModel';

export function Profile() {
  const props = useProfileViewModel();
  return <ProfileView {...props} />;
}
