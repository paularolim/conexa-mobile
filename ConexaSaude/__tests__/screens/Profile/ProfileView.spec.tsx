import renderer from 'react-test-renderer';

import { ProfileView } from '@screens/Profile';

import { Wrapper } from '../../Wrapper';

describe('ProfileView', () => {
  it('should render screen correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <ProfileView name="Dr. John Dow" email="johndoe@mail.com" handleLogout={jest.fn()} />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
