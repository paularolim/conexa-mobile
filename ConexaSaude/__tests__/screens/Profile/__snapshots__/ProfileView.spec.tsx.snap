// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileView should render screen correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "gap": 24,
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "gap": 16,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#C2CFE0",
            "borderRadius": 40,
            "height": 80,
            "overflow": "hidden",
            "width": 80,
          },
        ]
      }
    >
      <svg
        data-file-name="SvgUser"
        height={80}
        stroke="#F28080"
        width={80}
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 22,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Dr. John Dow
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
            },
          ]
        }
      >
        johndoe@mail.com
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "gap": 10,
        },
      ]
    }
  >
    <View
      primary={true}
      style={
        [
          {
            "backgroundColor": "#F28080",
            "borderRadius": 2,
            "height": 5,
            "width": 80,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#C2CFE0",
            "borderRadius": 2,
            "height": 5,
            "width": 80,
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "gap": 8,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "opacity": 1,
        }
      }
    >
      <svg
        data-file-name="SvgSignOut"
        height={32}
        stroke="#000000"
        width={32}
      />
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "fontSize": 18,
            },
          ]
        }
      >
        Sair
      </Text>
    </View>
  </View>
</View>
`;
